### DISCUSSION TITLE
How can I set middleware.js for a /[id]

### DISCUSSION THREAD

**Discussion by aly8i** [vercel/vercel#9096] [Help] [answered]
if I have a URL that is /products/[id] for example. How can I get the value of the [id] from the URL and what do I write in the url.includes()???

```
import {NextResponse} from "next/server";
import {jwtVerify} from 'jose';
export default async function middleware(req) {
    const token =  req.cookies.get('accessToken');
    const url = req.url;
    var p = {}
    if(url.includes("/admin")){
        try{
            const {payload} = await jwtVerify(token, new TextEncoder().encode(process.env.NEXT_PUBLIC_JWT_SECRET));
            p=payload
        }catch(e){
            return NextResponse.redirect(`${process.env.NEXT_PUBLIC_BASE_URL}`);
        }
        if(p.role =='admin'){
            return NextResponse.next();
        }
        return NextResponse.redirect(`${process.env.NEXT_PUBLIC_BASE_URL}`);
    }
        return NextResponse.next();
    
}
```

**Comment by amyegan** [comment, marked as answer]
Since you're [matching paths based on conditional statements](https://vercel.com/docs/concepts/functions/edge-middleware#match-paths-based-on-conditional-statements), you could check if `url.pathname.startsWith('/products/')`. Then you could use the full path to get the `id` value. 

You can find more detailed information in the [Vercel Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) and [Next.js Middleware](https://nextjs.org/docs/advanced-features/middleware) docs. The [NextRequest](https://nextjs.org/docs/api-reference/next/server#nextrequest) reference guide is also useful for reference.
